qiskit/algorithms/phase_estimators/hamiltonian_phase_estimation.py

```diff
@@ -88,49 +97,71 @@ class HamiltonianPhaseEstimation:
         """
+        if quantum_instance is not None:
+            warnings.warn(
+                "The quantum_instance argument has been superseded by the sampler argument. "
+                "This argument will be deprecated in a future release and subsequently "
+                "removed after that.",
+                category=PendingDeprecationWarning,
+            )
         self._phase_estimation = PhaseEstimation(
-            num_evaluation_qubits=num_evaluation_qubits, quantum_instance=quantum_instance
+            num_evaluation_qubits=num_evaluation_qubits,
+            quantum_instance=quantum_instance,
+            sampler=sampler,
         )
 
-    def _get_scale(self, hamiltonian, bound=None) -> None:
+    def _get_scale(self, hamiltonian, bound=None) -> PhaseEstimationScale:
         if bound is None:
             return PhaseEstimationScale.from_pauli_sum(hamiltonian)
 
         return PhaseEstimationScale(bound)
 
-    def _get_unitary(self, hamiltonian, pe_scale, evolution) -> QuantumCircuit:
+    def _get_unitary(
+        self, hamiltonian, pe_scale, evolution: EvolutionSynthesis | EvolutionBase
+    ) -> QuantumCircuit:
         """Evolve the Hamiltonian to obtain a unitary.
 
         Apply the scaling to the Hamiltonian that has been computed from an eigenvalue bound
         and compute the unitary by applying the evolution object.
         """
-        # scale so that phase does not wrap.
-        scaled_hamiltonian = -pe_scale.scale * hamiltonian
-        unitary = evolution.convert(scaled_hamiltonian.exp_i())
-        if not isinstance(unitary, QuantumCircuit):
-            unitary_circuit = unitary.to_circuit()
+
+        if self._phase_estimation._sampler is not None:
+
+            evo = PauliEvolutionGate(hamiltonian, -pe_scale.scale, synthesis=evolution)
+            unitary = QuantumCircuit(evo.num_qubits)
+            unitary.append(evo, unitary.qubits)
+
+            return unitary.decompose().decompose()
         else:
-            unitary_circuit = unitary
+            # scale so that phase does not wrap.
+            scaled_hamiltonian = -pe_scale.scale * hamiltonian
+            unitary = evolution.convert(scaled_hamiltonian.exp_i())
+            if not isinstance(unitary, QuantumCircuit):
+                unitary = unitary.to_circuit()
+
+            return unitary.decompose().decompose()
 
         # Decomposing twice allows some 1Q Hamiltonians to give correct results
         # when using MatrixEvolution(), that otherwise would give incorrect results.
         # It does not break any others that we tested.
-        return unitary_circuit.decompose().decompose()
 
     def estimate(
         self,
-        hamiltonian: OperatorBase,
-        state_preparation: Optional[StateFn] = None,
-        evolution: Optional[EvolutionBase] = None,
-        bound: Optional[float] = None,
+        hamiltonian: OperatorBase | BaseOperator | PauliSumOp,
```

Comment:
This added BaseOperator (OperatorBase is opflow of which PauliSumOp, also added, is a specific type) With quantum instance the original code path supported certain opflow operators; what is the case with BaseOperator? I get the latter spec just added `BaseOperator | PauliSumOp` is what primitives handle. Here also a PauliEvolutionGate appears used - that seems to take a specific set of operators. The orginal typehint for OperatorBase was arguably too broad and could have stated the 4 types supported as per instance checks from line 216. Do we need more checking for new Sampler path or it is good for all BaseOperator?